risc0/build/src/lib.rs

```diff
@@ -45,13 +45,11 @@ impl Risc0Metadata {
 struct Risc0Method {
     name: String,
     elf_path: PathBuf,
+    cache_dir: PathBuf, // It's a bit questionable if this belongs here
 }
 
 impl Risc0Method {
```

Comment:
@shkoo See here. Instead of writing the hash to `method_id.elf_sha`, I wrote the method id to `{hash}.cache`

Reply:
I'm actually not sure why this was changed actually. The thing that we compare as to whether we need to re-compute the MethodID is the SHA of the ELF. It seems like the only change needed in this file is just the call to `make_method_id_from_elf` instead of `MethodID::new`. Would be even nicer if we kept the same `MethodID::new` interface.

Reply:
To be clear, we need to generate 3 files:
* The ELF file itself
* The SHA of the ELF (for later detection of the need to rebuild MethodID).
* The cached MethodID (so we can reuse it if the SHA hasn't changed)

Reply:
I think I can see why this logic will work now. However, the one issue I see with this new approach is that the target dir will fill up with stale `{hash}.cache` paths that don't go away until someone cleans their target dir.

Reply:
Yeah the multiple files solution is definitely better because of the lack of pollution. But re: your first comment I had to get the method_id from somewhere. I could've made a different file, put it in the same file and parsed it out, or did what i did. I probably chose the wrong solution and we should fix it. I'm not sure I see the reason why the MethodID::new interface would've been better. I found having the additional type to be a bit confusing because we still had to deal with the byte array no matter what. :shrug: